docs/proposals/policy.md

```diff
@@ -0,0 +1,251 @@
+  1.  exec
+  1.  watch
+  1.  list
+  1.  get
+  1.  create
+  1.  update
+  1.  delete
+  1.  proxy
+
+### Verb Specific Attributes
+Different verbs will require different attributes.  For instance, a list would have a kind, a get would have an kind and an id, and an update may have a kind, an id, and a list of the fields being modified.  Although exec isn't well defined, it could very well have an entirely distinct set of attributes.  The exact contract rules will be chosen by the Authorizer/AuthorizationAttributeBuilder pair, but basic resource access for openshift should include kind in order to be compatible with existing kubernetes authorization.
+
+### Configtime Authorization Types
+```
+// PolicyRule holds information that describes a policy rule, but does not contain information 
+// about who the rule applies to or which namespace the rule applies to.  This makes PolicyRules portable units.
+type PolicyRule struct {
+	// Deny is true if any request matching this rule should be denied.  If false, any request matching this rule is allowed.
+	Deny bool
+	// Verbs is a list of Verbs that apply to ALL the ResourceKinds and AttributeRestrictions contained in this rule
+	Verbs []Verb
+	// AttributeRestrictions will vary depending on what the Authorizer/AuthorizationAttributeBuilder pair supports.  
+	// Making it an interface{} allows a type switch for identification followed by properly typed usage thereafter.
+	AttributeRestrictions interface{}
+	// ResourceKinds is a list of kinds this rule applies to
+	ResourceKinds []string
+}
+
+// PolicyRuleBinding builds on top of a PolicyRule.  It adds who information (usernames, teamnames, accountnames) 
+// and namespace information (ObjectMeta).  PolicyRuleBindings only apply to the namespace they are contained in.  
+// That means the full set of PolicyRuleBindings for a namespaced request are contained inside of the requested 
+// namespace or in the global namespace
+type PolicyRuleBinding struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	UserNames    []string
+	TeamNames    []string
+	AccountNames []string
+
+	Rule PolicyRule
+}
+
+// Role is a logical grouping of PolicyRules that can be referenced as a unit by RoleBindings.
+type Role struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	Rules []PolicyRule
+}
+
+// RoleBinding references a Role, but not contain it.  It adds who and namespace information like a PolicyRuleBinding.  
+// It can reference any Role in the same namespace or in the global namespace.
+type RoleBinding struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	UserNames    []string
+	TeamNames    []string
+	AccountNames []string
+
+	// this is a ref to a Role
+	ReferencesGlobalRole bool
```

Comment:
What happens if the global role is deleted?  Are we doing anything to prevent deletion of a global role if its in use in a more local context?  If so, how.  If not, what happens.